### ISSUE TITLE
Wii Port - D-Pad Up Key doesn't work in menu when selecting a game

### ISSUE THREAD

**Issue by SieKensou** [DCurrent/openbor#69]
I have tested the last version (thanks for fixing it to make it work with USB) and it works fine with Game Cube Controller or Wii-Mote controller, but when using my original Pro Controller it doesn't work when I press up with the D-pad (it works if I use the analog, but not in the D-Pad). It works perfect in-game too, only doesn't work Up in the menu when selecting between the diferent pak files.

**Comment by msmalik681** [comment]
That is a odd bug @demothens can you check why this is not working on the pak selection menu as you added the support for the pro pad and can you use the trunk version as I added the support using the lib and include files in the tools folder and I did not use that wupc.c file just the wupc.h as recommended on the github for the fix.

**Comment by SieKensou** [comment]
It's the Classic Pro Controller for the Wii, not the Wii U Pro Controller. I don't have the standard Classic controller at hand right now to check if it happens with that one too, but it should as they are the same in functionability.
The one I have tested is this one (Original from Nintendo): https://upload.wikimedia.org/wikipedia/commons/thumb/4/43/Gold_classic_controller_pro.JPG/800px-Gold_classic_controller_pro.JPG

**Comment by Wiimpathy** [comment]
It doesn't work with the Wii Classic controller either. In refreshInput() from menu.c you should seperate the classic buttons. Maybe like this:

```
	if(wpad->exp.type <= WPAD_EXP_NUNCHUK)
	{
		if(wpad->btns_h & WPAD_BUTTON_1)                      btns |= WIIMOTE_1;
		if(wpad->btns_h & WPAD_BUTTON_2)                      btns |= WIIMOTE_2;
		if(wpad->btns_h & WPAD_BUTTON_A)                      btns |= WIIMOTE_A;
		if(wpad->btns_h & WPAD_BUTTON_B)                      btns |= WIIMOTE_B;
		if(wpad->btns_h & WPAD_BUTTON_MINUS)                  btns |= WIIMOTE_MINUS;
		if(wpad->btns_h & WPAD_BUTTON_PLUS)                   btns |= WIIMOTE_PLUS;
		if(wpad->btns_h & WPAD_BUTTON_HOME)                   btns |= WIIMOTE_HOME;
		if(wpad->btns_h & WPAD_NUNCHUK_BUTTON_Z)              btns |= NUNCHUK_Z;
		if(wpad->btns_h & WPAD_NUNCHUK_BUTTON_C)              btns |= NUNCHUK_C;
	}
	else if(wpad->exp.type == WPAD_EXP_CLASSIC)
	{
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_A)              btns |= CC_A;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_B)              btns |= CC_B;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_Y)              btns |= CC_Y;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_X)              btns |= CC_X;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_MINUS)          btns |= CC_MINUS;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_PLUS)           btns |= CC_PLUS;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_HOME)           btns |= CC_HOME;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_FULL_R)         btns |= CC_R;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_FULL_L)         btns |= CC_L;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_ZL)             btns |= CC_ZL;
		if(wpad->btns_h & WPAD_CLASSIC_BUTTON_ZR)             btns |= CC_ZR;
	}

```

**Comment by msmalik681** [comment]
@Wiimpathy thanks i will make a test build with this change.

**Comment by Wiimpathy** [comment]
I don't think so. What's been fixed recently was a bug in-game: https://github.com/DCurrent/openbor/commit/fdcb7a6ab70b72aae8b342137f6125ad3bc1e1cc
Here, it only happens in the menu with specific Wii controllers. There's a possible fix in this branch : https://github.com/Wiimpathy/openbor/commits/wii-plugin-and-fixes

**Comment by Plombo** [comment]
I'm still away from home on holiday travel right now, so it will be at least a couple more days before I can look at it.

**Comment by msmalik681** [comment]
I can confirm @Wiimpathy your updates have fixed any controller issues could you please add your wii updates to the master build.

Also I noticed you made changes so we can use DevkitPPC r29+ I heard there are issues with later builds if everything is working is it worth updating ?

**Comment by Wiimpathy** [comment]
Since R28, Wii-U pro controller are supported natively, no need of libwupc. Not sure it's better, I never tested...
R30+ have a newer gcc. It seems performance aren't as good as gcc 6. Never done real benchmark though.
One thing is sure, the dol files are bigger with recent libs : less Ram!
Changelogs:
http://nintendo-ds.dcemu.co.uk/devkitppc-r28-and-libogc-v1-8-15-released-wii-u-development-1158311.html
http://nintendo-ds.dcemu.co.uk/devkitppc-r29-released-kitchen-sink-update-1158398.html
See if it's worth it.

2d405a27cfeafc77ba9cb84ad890100ffab943a0 has already been partially merged. 
As for argument's support I don't  know if you're interested...

I don't mind if you copied my changes directly and pick what you want, it may be simpler.

**Comment by msmalik681** [comment]
@Wiimpathy thanks I will add the update crediting you in the commit.

Less ram is a definite no so we will stick to the current DevkitPPC.
